File: AdminService/src/main/java/com/ecommercial/shopping/adminservice/global/database/ReadOnlyDataSourceCycle.java
package com.ecommercial.shopping.adminservice.global.database;

import lombok.Setter;

import java.util.List;

@Setter
public class ReadOnlyDataSourceCycle<T> {
    private List<T> readOnlyDataSourceLookupKeys;
    private int index = 0;

    public T getReadOnlyDataSourceLookupKey() {
        if (index + 1 >= readOnlyDataSourceLookupKeys.size()) {
            index = -1;
        }
        return readOnlyDataSourceLookupKeys.get(++index);
    }

}

File: UserService/src/main/kotlin/com/ecommercial/shopping/userservice/global/database/ReadOnlyDataSourceCycle.kt
package com.ecommercial.shopping.userservice.global.database

import lombok.Setter

@Setter
class ReadOnlyDataSourceCycle<T> {
    var readOnlyDataSourceLookupKeys : List<T> = emptyList()
    private var index: Int = 0

    fun getReadOnlyDataSourceLookupKey(): T {
        if (index + 1 >= readOnlyDataSourceLookupKeys.size) {
            index = -1
        }
        index += 1
        return readOnlyDataSourceLookupKeys[index]
    }
}